bounce/server/api/clubs.py

```diff
@@ -33,30 +35,53 @@ class ClubEndpoint(Endpoint):
             raise APIError('No such club', status=404)
         return response.json(club_data, status=200)
 
+    @verify_token()
     @validate(PutClubRequest, GetClubResponse)
-    async def put(self, request, name):
+    async def put(self, request, name, id_from_token=None):
         """Handles a PUT /clubs/<name> request by updating the club with
         the given name and returning the updated club info."""
         # Decode the name, since special characters will be URL-encoded
         name = unquote(name)
         body = util.strip_whitespace(request.json)
-        updated_club = club.update(
-            self.server.db_session,
-            name,
-            new_name=body.get('name', None),
-            description=body.get('description', None),
-            website_url=body.get('website_url', None),
-            facebook_url=body.get('facebook_url', None),
-            instagram_url=body.get('instagram_url', None),
-            twitter_url=body.get('twitter_url', None))
+        try:
+            editor_attr = membership.select(self.server.db_session, name,
+                                            id_from_token,
+                                            Roles.president.value)
+            editors_role = editor_attr[0]['role']
+            updated_club = club.update(
+                self.server.db_session,
+                name,
+                editors_role,
+                new_name=body.get('name', None),
+                description=body.get('description', None),
+                website_url=body.get('website_url', None),
+                facebook_url=body.get('facebook_url', None),
+                instagram_url=body.get('instagram_url', None),
+                twitter_url=body.get('twitter_url', None))
+        except PermissionError:
+            raise APIError('Unauthorized', status=403)
         return response.json(updated_club, status=200)
 
-    async def delete(self, _, name):
+    @verify_token()
+    @validate(DeleteClubRequest, None)
+    async def delete(self, request, name, id_from_token=None):
```

Comment:
To fix the linting problem you can just change the name `request` in the function header to `_`.